csrc/device_lower/pass/index.cpp

```diff
@@ -1479,8 +1520,21 @@ void IndexLowering::handle(const PadOp* pad) {
 
   const auto pad_val = pad->value();
 
+  std::unordered_map<IterDomain*, Val*> override_index;
+  for (auto padded_axis : pad->getPaddedAxes()) {
+    auto padded_id = producer_doms.at(padded_axis);
+    if (padded_id->isBroadcast()) {
+      // When we pad a Broadcast IterDomain, we should not treat it as a
```

Comment:
Trying to understand why. Is this only necessary for the padding predicate? For the normal indexing, we shouldn't need to index into the broadcast domain, right?

Reply:
Yes it's just for the padding predicate. Otherwise we will have a constant 0 which means the padding predicate is always true, so we never use the pad value.

Reply:
I wonder if we could minimize the usage of `override_index`. That's a workaround for ops like `index_select`, and it shouldn't be used unless absolutely necessary.

Since it should result in `0`, we may stop kipping broadcast domains. I believe that should give us a correct predicate for padding, and we still end up having the same predicates and indices in anywhere else.

Reply:
One correction. It's already using `override_index` only for generating the pad predicate.

Reply:
Sorry, my comment above was completely wrong. Still trying to get a better understanding...